src/main/java/com/marklogic/kafka/connect/source/QueryHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.RowBatcher;
+import org.apache.kafka.common.config.ConfigException;
+import org.springframework.util.StringUtils;
+
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public interface QueryHandler {
+    void addQueryToRowBatcher(RowBatcher<?> rowBatcher);
+
+    String updatePreviousMaxConstraintColumnValue(long queryStartTimeInMillis);
+
+    String ANY_QUOTES_REGEX_PATTERN = "[\"']";
+    default String sanitize(String target) {
+        return target.replaceAll(ANY_QUOTES_REGEX_PATTERN, "");
+    }
+
+    String WHITESPACE_REGEX_PATTERN = "\\s+(?=(?:[^\\'\"]*[\\'\"][^\\'\"]*[\\'\"])*[^\\'\"]*$)";
+    default String removeWhitespace(String target) {
+        return target.replaceAll(WHITESPACE_REGEX_PATTERN, "");
+    }
+
+    static QueryHandler newQueryHandler(DatabaseClient databaseClient, Map<String, Object> parsedConfig) {
+        boolean configuredForDsl = StringUtils.hasText((String) parsedConfig.get(MarkLogicSourceConfig.DSL_QUERY));
+        boolean configuredForSerialized = StringUtils.hasText((String) parsedConfig.get(MarkLogicSourceConfig.SERIALIZED_QUERY));
+        QueryHandler queryHandler;
+        if ((!(configuredForDsl || configuredForSerialized)) || (configuredForDsl && configuredForSerialized)) {
+            throw new ConfigException(
+                format("Either a DSL Optic query (%s) or a serialized Optic query (%s), but not both, are required",
+                    MarkLogicSourceConfig.DSL_QUERY, MarkLogicSourceConfig.SERIALIZED_QUERY)
+            );
+        }
+        if (configuredForDsl) {
```

Comment:
I think this is where the ternary operator can cut down on some Java verbosity - i.e. you can avoid the queryHandler declaration above and just do :

```
return configuredForDsl ? 
    new Dsl... : 
    new Serialized ...
```

I almost always go for that instead of declaring a variable for the sole purpose of returning it.